packages/typescript-estree/src/create-program/createIsolatedProgram.ts

```diff
@@ -65,6 +65,9 @@ function createIsolatedProgram(
   const program = ts.createProgram(
     [parseSettings.filePath],
     {
+      ...(ts.JSDocParsingMode && {
+        jsDocParsingMode: ts.JSDocParsingMode.ParseForTypeInfo,
```

Comment:
We should use your benchmarking tooling to do a before and after with this line to see how much it has improved the perf.

My original perf profiling way back when showed a decent chunk of time spent but things may have changed since then.
Note: the more important time would be the non-type-aware lint time as type info would dwarf any savings we get.

Reply:
👍 filed #7906